discrete_math/representations_of_numbers/representation.hpp

```diff
@@ -0,0 +1,147 @@
+#include <iostream>
+#include <concepts>
+#include <variant>
+#include <expected>
+#include <variant>
+#include <vector>
+#include <string>
+
+namespace representation_namespace {
+
+    const int NUMBERS = 48;
+    const int CHARS = 55;
+
+    template <std::integral I>
+    struct fraction {
+        I numerator{ 0 };
+        I denominator{ 1 };
+
+        fraction(I num = 0, I den = 1) : numerator(num), denominator(den) {}
+    };
+
+    template <std::uint8_t base>
+    struct expansion {
+        static constexpr std::uint8_t form = base;
+    };

```

Comment:
```suggestion
    template <std::uint8_t base>
    struct expansion {
        static constexpr std::uint8_t base_v = base;
    };

```

Reply:
stick to the api. Add to expansion the following members
std::string whole;
std::string fractional;
std::string period;
then the expansion represents expansions of a fraction of the form
whole.fractional(period)